app/controllers/api/v2/users_controller.rb

```diff
@@ -31,11 +33,29 @@ def accept_terms
       def generate_and_send_magic_link
         skip_authorization
 
-        GenerateAndSendMagicLinkService.call(user: current_user)
+        user = User.where(email: email_from_token).first
+
+        if user
+          GenerateAndSendMagicLinkService.call(user: user)
+        else
+          NotificationService::UnrecognisedEmail.call(email: email_from_token)
+        end
       end
 
     private
 
+      def email_from_token
```

Comment:
lacks validation

Reply:
Not sure I understand the change you want here 🤔 
Are you saying the email address needs to be validated?

Reply:
Will add email validation to Publish in separate PR @defong